src/Azure.Functions.Cli/Kubernetes/KEDA/KedaResourceFactory.cs

```diff
@@ -0,0 +1,86 @@
+﻿using Azure.Functions.Cli.Kubernetes.KEDA.V1;
+using Azure.Functions.Cli.Kubernetes.KEDA.V2;
+using Azure.Functions.Cli.Kubernetes.Models;
+using Azure.Functions.Cli.Kubernetes.Models.Kubernetes;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Azure.Functions.Cli.Kubernetes.KEDA
+{
+    public class KedaResourceFactory
```

Comment:
I can see why you're proposing this, and agree on the Create, but `PopulateMetadataDictionary` & `GetKedaTrigger` are different per KEDA version and that's why I went with dedicated factories.

So what I would propose is that we keep `KedaV1ResourceFactory` & `KedaV2ResourceFactory` which uses this as a base for Create or enforce it with an interface, but the rest is version specific and handled in the dedicated factories.

Ok for you?

Reply:
I recommed - Single factory  `KedaResourceFactory` that creates KedaResource with the version chosen. And yes, it is better to have an interface `IKedaResource`  -  If there is any commonality between versions - also create a base class.